src/Hamiltonians/FroehlichPolaron.jl

```diff
@@ -137,7 +137,7 @@ function get_offdiagonal(h::FroehlichPolaron{T,M,<:Any,T}, addr::OccupationNumbe
     naddress, value = _froehlich_offdiag(h, addr, chosen)
 
     new_p_tot = dot(h.ks, onr(naddress))
-    if (M/h.l) * new_p_tot > h.momentum_cutoff # check if momentum of new address exceeds momentum_cutoff
+    if abs(new_p_tot) > h.momentum_cutoff # check if momentum of new address exceeds momentum_cutoff
         return addr, 0.0
     else
         return naddress, - h.v * value
```

Comment:
```suggestion
        return naddress, value
```
Since we already multiply by `v` in `_froehlich_offdiag` we don't need to again here.

Reply:
OMG, you found another bug.